src/daemon.py

```diff
@@ -0,0 +1,93 @@
+import os
+import time
+import pandas as pd
+from web3 import Web3
+from typing import List
+from threading import Thread
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+from src.imbalances_script import RawTokenImbalances
+from src.config import CHAIN_RPC_ENDPOINTS, CHAIN_SLEEP_TIMES
+
+def get_web3_instance(chain_name: str) -> Web3:
+    return Web3(Web3.HTTPProvider(CHAIN_RPC_ENDPOINTS[chain_name]))
+
+def get_finalized_block_number(web3: Web3) -> int:
+    return web3.eth.block_number - 64
+
+def create_db_connection(chain_name: str):
+    """function that creates a connection to the CoW db.""" 
+    if chain_name == 'Ethereum':
+        prod_url = os.getenv("ETHEREUM_DB_URL")
+    elif chain_name == 'Gnosis':
+        prod_url = os.getenv("GNOSIS_DB_URL")
+
+    return create_engine(f"postgresql+psycopg2://{prod_url}")
+
+def fetch_transaction_hashes(db_connection: Engine, start_block: int, end_block: int) -> List[str]:
+    """Fetch transaction hashes beginning start_block."""
+    query = f"""
+    SELECT tx_hash 
+    FROM settlements 
+    WHERE block_number >= {start_block}
+    AND block_number <= {end_block}
+    """
+
+    prod_hashes = pd.read_sql(query, db_connection)
```

Comment:
Same comment for the "prod" prefix. This is just a tx hash from the point of view of the script